test/Kestrel.FunctionalTests/UpgradeTests.cs

```diff
@@ -297,5 +297,42 @@ public class UpgradeTests : LoggedTest
             var exception = await Assert.ThrowsAsync<InvalidOperationException>(async () => await upgradeTcs.Task.TimeoutAfter(TimeSpan.FromSeconds(60)));
             Assert.Equal(CoreStrings.UpgradedConnectionLimitReached, exception.Message);
         }
+
+        [Fact]
+        public async Task DoesNotThrowOnFin()
+        {
+            var appCompletedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            using (var server = new TestServer(async context =>
+            {
+                var feature = context.Features.Get<IHttpUpgradeFeature>();
+                var duplexStream = await feature.UpgradeAsync();
+
+                try
+                {
+                    await duplexStream.CopyToAsync(Stream.Null);
+                    appCompletedTcs.SetResult(null);
+                }
+                catch (Exception ex)
+                {
+                    appCompletedTcs.SetException(ex);
+                    throw;
+                }
+
+            }, new TestServiceContext(LoggerFactory)))
```

Comment:
Could you change the formatting to make it obvious that the `using(var server) {` starts here?

Reply:
I'm keeping this consistent with the other tests in this class and Kestrel in general. If you think there's better formatting, feel free to submit a PR, but it's will be a big diff.